EngineLayer/ClassicSearch/ClassicSearchEngine.cs

```diff
@@ -89,22 +100,28 @@ protected override MetaMorpheusEngineResults RunSpecific()
                         // digest each protein into peptides and search for each peptide in all spectra within precursor mass tolerance
                         foreach (PeptideWithSetModifications peptide in Proteins[i].Digest(CommonParameters.DigestionParams, FixedModifications, VariableModifications, SilacLabels, TurnoverLabels))
                         {
-                            foreach (var fragmentSet in fragmentsForDissociationTypes)
+                            PeptideWithSetModifications reversedOnTheFlyDecoy = null;
+
+                            if (SpectralLibrary != null)
+                            {
+                                int[] newAAlocations = new int[peptide.BaseSequence.Length];
+                                reversedOnTheFlyDecoy = peptide.GetReverseDecoyFromTarget(newAAlocations);
+                            }
+
+                            // clear fragments from the last peptide
+                            foreach (var fragmentSet in targetFragmentsForEachDissociationType)
                             {
                                 fragmentSet.Value.Clear();
+                                decoyFragmentsForEachDissociationType[fragmentSet.Key].Clear();
```

Comment:
I don't see when this is ever populated.

Reply:
It is populated in line 86,92,155.